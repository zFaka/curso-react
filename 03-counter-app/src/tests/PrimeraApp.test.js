import React from 'react';
import PrimeraApp from '../PrimeraApp';
import {shallow} from 'enzyme';

describe('PrimeraApp test', () => {

    //test('Should return a "Faketah" message', () => {
    //const saludo = 'Faketah'; 
    //const {getByText} = render(<PrimeraApp saludo={saludo}/>);
    //expect(getByText(saludo)).toBeInTheDocument();
    //})

    test('Should show <PrimeraApp/> correctly', () => {

        const saludo = 'Faketah'; 
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);

        expect(wrapper).toMatchSnapshot()

    })

    test('Should show subtitle sended by props', () => {

        const saludo = 'FAKA'; 
        const subTitulo = 'First Application'; 

        const wrapper = shallow(
            <PrimeraApp 
                saludo={saludo} 
                subtitulo={subTitulo}
            />);

        //Si fuese la busqueda por id seria .find('#idQueSeBusca')
        //Si fuese la busqueda por clases seria .find('.claseQueSeBusca')
        const textoSubtitulo = wrapper.find('p').text(); //Extrae texto dentro de p

        expect(textoSubtitulo).toBe(subTitulo)

    })

})
