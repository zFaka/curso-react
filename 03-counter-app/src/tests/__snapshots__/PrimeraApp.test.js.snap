// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PrimeraApp test Should show <PrimeraApp/> correctly 1`] = `
<Fragment>
  <h1>
    Faketah
  </h1>
  <p>
    First App
  </p>
</Fragment>
`;
